README.md

```diff
@@ -3,53 +3,112 @@
-* Web Server (Apache)
+* Web Server (nginx)
 
-The ansible playbook(s) are written for RHEL7 based distributions (CentOS, SciLi, etc), but it probably wouldn't be too hard to extend the playbooks for Debian based distros.
+The ansible playbook is written for RHEL7 based distributions (CentOS, SciLi, etc) and Ubuntu 16.04.  It _may_ work for Ubuntu 14.04 and RHEL6, but there are a few systemd dependent tasks.
 
 ## Roles/Profiles
 The three application components are divided in Ansible roles:
-* [db](roles/db)
-* [app](roles/app)
-* [web](roles/web)
+* [db](roles/db) 
+* [app](roles/app) 
+* [web](roles/web) 
 
+Both postgres and nginx are installed from the "upstream" apt/yum repos (instead of the base CentOS or Ubuntu repos).  This provides a semi-consistent across operating systems.
 
+There is a fourth [base](roles/base) role that applies to _all_ hosts.  This base role includes stuff like ntp, certbot, additional ssh keys, etc.
 
+## Networking
+All `WEB <-> APP <-> DB` traffic occurs over the network using TLS v1.2.  Firewalld is used to expose the DB and uWSGI to (only) the application servers and web servers respectively.
 
-A sample [playbook](single-host.yml) is included as an example. This playbook configures all three services to run on one host.
+By default, all application servers will connect to the **first** database server listed in the `[db]` ansible group.  The current deployment is designed to utilize a single postgres DB instance.  Multiple DB servers can be deployed, however, only the first DB server will be used by the application servers.
 
-Sensitive information (http private keys, application API keys) are stored in ansible-vault files in their respective roles directory.
 
-Sever specific configuration items are included in: `roles/TARGETROLE/vars/SERVERHOSTNAME`
+## Inventory File
 
-As an example, the http certs and keys for test.theoval.us are stored in [roles/web/vars/test.theoval.us.yml](roles/web/vars/test.theoval.us.yml)
+The default `site.yml` playbook will apply each role to the similarly named ansible group.  E.g. any hosts in the `[db]` ansible group will receive the `db` role.  An example inventory file with 7 hosts:
+ * 1 x CentOS 7 database (postgres) server
+ * 2 x Ubuntu 16.04 application (uwsgi) servers
+ * 2 x CentOS 7 web (nginx) servers
+ * 2 x Ubuntu 16.04 web (nginx) servers
+```
+[db]
+db1.example.com	ansible_user=root 
+
+[app]
+app1.example.com ansible_user=root ansible_python_interpreter=/usr/bin/python3
+app2.example.com ansible_user=root ansible_python_interpreter=/usr/bin/python3
+
+[web]
+web1.example.com ansible_user=root 
+web2.example.com ansible_user=root 
+web3.example.com ansible_user=root ansible_python_interpreter=/usr/bin/python3
+web4.example.com ansible_user=root ansible_python_interpreter=/usr/bin/python3
+```
+
+Any servers that use python3 by default (e.g. Ubuntu 16.04) **must** have the `ansible_python_interpreter` variable set to `/usr/bin/python3`.
```

Comment:
Ansible (_mostly_) supports python2 and python3.  However, ansible will assume a python interpreter named `python` is available in your `$PATH`. For distributions like Ubuntu 16.04, the default python interpreter is `/usr/bin/python3` (python 2 is **not** installed by default).

So you have two options:

- Log in before Ansible runs and manually install python2
- Set the location of your python interpreter (typically in the inventory file) to `/usr/bin/python3`